test/DualTest.jl

```diff
@@ -455,4 +455,14 @@ for N in (0,3), M in (0,4), V in (Int, Float32)
     end
 end
 
+@testset "Exponentiation of zero" begin
```

Comment:
I thought to mix this with the rest of the test but then can't think of a good way to turn this into a random test.  So I created a separated testset.  I also noticed that `@testset` is not used anywhere but I assumed that's only because ForwardDiff.jl predates `@testset`.  I hope using `@testset` is OK.